### PR TITLE
SQL audit index results

### PR DESCRIPTION
**Pull request by snopoke**
Details of changes here: https://docs.google.com/spreadsheets/d/1DWg9icG-7BVTxAlnWlzdOQaekxp4m9Jz9YKZZi3P1TM/edit#gid=1873309260

@orangejenny 
@dimagi/scale-team I'd appreciate a second opinion if anyone has time

### CODE REVIEW

File: corehq/form_processor/models.py [hunk 1]
```diff
@@ -365,7 +365,6 @@ class Meta:
 
 class AbstractAttachment(DisabledDbMixin, models.Model, SaveStateMixin):
     attachment_id = UUIDField(unique=True, db_index=True)
-    name = models.CharField(max_length=255, db_index=True, default=None)
```
Comment (czue) [inline corehq/form_processor/models.py]: kind of annoying that you can't keep this in the base class if you want to index differently (not proposing anything, just venting)

File: corehq/form_processor/models.py [hunk 2]
```diff
@@ -1218,14 +1218,14 @@ class LedgerValue(DisabledDbMixin, models.Model, TrackRelatedChanges):
     objects = RestrictedManager()
 
     domain = models.CharField(max_length=255, null=False, default=None)
-    case_id = models.CharField(max_length=255, db_index=True, default=None)  # remove foreign key until we're sharding this
+    case_id = models.CharField(max_length=255, default=None)  # remove foreign key until we're sharding this
```
Comment (czue) [inline corehq/form_processor/models.py]: is this comment still relevant? do we shard these by `case_id`? any reason this isn't a foreign key? (mostly asking for my own education)

**Comment by czue** [comment]
looks good to me. lots of real looking test failures

File: corehq/form_processor/models.py (hunk 2 above)
Comment (snopoke) [inline corehq/form_processor/models.py]: This is sharded by `case_id` so we could make it a foreign key which I guess we should. I'll update.

**Comment by snopoke** [comment]
:white_check_mark:

File: corehq/form_processor/models.py (hunk 2 above)
Comment (czue) [inline corehq/form_processor/models.py]: (assume you'll do this at some point later if you think worth it)

Comment (snopoke) [inline corehq/form_processor/models.py]: https://github.com/dimagi/commcare-hq/pull/12654

**Pull request merged**